### ISSUE TITLE
Consider using Slurm's sbcast command to move files in and out of sandboxes?

### ISSUE THREAD

**Issue by rhliang** [cfe-lab/Kive#640]
Already seeing some evidence that file copying issues (#569, #618) will affect the Slurm branch.  

Some context, first.  The current execution scheme is that each step progresses in three stages: setup; driver; and bookkeeping.  The setup stage runs all the cables and puts the required code in place.  The driver is a straight call to `sbatch` telling it to invoke the code on the specified outputs, write the outputs to the specified paths, etc.  Bookkeeping handles everything after that.

I'm running the simple sums and products pipeline on my Slurm branch, and found that at the driver stage, `sbatch` regularly complained that the driver wasn't accessible.  When I did a little probing, I found that the driver wasn't put into place by the setup stage of the execution.  When I built a 5-second sleep into the Slurm submission method, it worked.

Note that this is all running in a virtual machine, so NFS isn't an issue.

I'm wondering if it would make sense as a long-term solution to transition away from our current methods of moving files around and start using Slurm's `sbcast` command to move files from point A to point B.  I suspect it might be more robust.  This would mean a bit of an overhaul.

**Comment by jamesnakagawa** [comment]
@rhliang, I pulled from master and now I'm having a `scandir` error that I think you mentioned on Monday regarding Slurm+OSX when I do `manage.py runserver`. Can you confirm this is related?

**Comment by rhliang** [comment]
No, it wouldn't be related, because the Slurm stuff is not in the master branch.

**Comment by rhliang** [comment]
I think @wrpscott mentioned that `scandir` was now a dependency though, so you'll have to install that.

**Comment by jamesnakagawa** [comment]
Ooh okay.

**Comment by wrpscott** [comment]
Hello James!

yes, scandir is a new dependency, it can be installed with pip...
Let me know if there are any problems with this...

kind regards,
Scotty

**Comment by jamesnakagawa** [comment]
I’ve got it working again, thanks Scotty.

**Comment by rhliang** [comment]
Discussed at meeting; this seems like a decent thing to explore post-0.9 (and it should help with issues such as #569).
